library/tiqr/tests/OcraTest.php

```diff
@@ -65,4 +65,38 @@ public function testChallengeResponseWithSession()
         $this->assertEquals("675831", $result);
     }
 
+    public function testTruncate() {
+        // Test RFC4226 truncate algorithm
+        // Use SHA-1 with an integer value to generate the strings to truncate
+        $tests=array(
+//          sha => expected result
+            6 => 37787,
+            23 => 24262,
+            114 => 5582,
+            203 => 3164,
+            480 => 385,
+            1784 => 313,
+            5001 => 65,
+            8012 => 12,
+            9536 => 41702,
+            22264 => 16058,
+            35619 => 3,
+            71453 => 5,
+            50979 => 16854,
+            68359 => 54266,
+            113274 => 7068,
+        );
+        foreach ($tests as $i => $expected) {
+            $hash=sha1($i);
+            $t=OCRA::_oath_truncate($hash, 6);
+            $this->assertIsString($t);
+            $this->assertTrue(strlen($t) == 6);
+
+            $this->assertEquals( $expected, (int)$t, "$i : $hash : t=$t");
+        }
+
+        // String consisting of "0000...000" must result in 0
+        $this->assertEquals( 0, (int)OCRA::_oath_truncate(str_repeat("00", 20), 6) );
```

Comment:
Again thanks for actively adding tests to this legacy code base!

The test itself looks very relevant. One remark about the process: our unit testing strategy would put the highlighted part of the test in a separate test method. As you are testing a different testcase. That way tests remain more focused on a single task.

Feel free to leave this as is for now!